src/gluonts/model/forecast_generator.py

```diff
@@ -241,3 +262,114 @@ def __call__(
                     info=batch["info"][i] if "info" in batch else None,
                 )
             assert i + 1 == len(batch[FieldName.FORECAST_START])
+
+
+class ForecastBatch:
+    @property
+    def batch_size(self) -> int:
+        raise NotImplementedError
+
+    @property
+    def mean(self) -> np.ndarray:
+        raise NotImplementedError
+
+    def quantile(self, q: Union[float, str]) -> np.ndarray:
+        raise NotImplementedError
+
+
+@dataclass
+class DistributionForecastBatch(ForecastBatch):
+    start_date: list
+    item_id: Optional[list]
+    info: Optional[list]
+    distr_output: Any  # TODO fix
+    distr_args: list
+
+    def __post_init__(self):
+        self.distr = self.distr_output.distribution(*self.distr_args)
```

Comment:
You would normally include `distr` in the declarations, but use `distr: Type = field(init=False)`